addons/stock/models/stock_move_line.py

```diff
@@ -1048,27 +1113,42 @@ def _post_put_in_pack_hook(self, package, **kwargs):
                 return action
         return package
 
-    def _to_pack(self):
+    def _to_pack(self, without_pack=True):
         if len(self.picking_type_id) > 1:
             raise UserError(_('You cannot pack products into the same package when they are from different transfers with different operation types'))
         quantity_move_line_ids = self.filtered(
-            lambda ml: ml.product_uom_id.compare(ml.quantity, 0.0) > 0 and not ml.result_package_id
+            lambda ml: ml.product_uom_id.compare(ml.quantity, 0.0) > 0 and (without_pack != bool(ml.result_package_id))
             and ml.state not in ('done', 'cancel')
         )
         move_line_ids = quantity_move_line_ids.filtered(lambda ml: ml.picked)
         if not move_line_ids:
             move_line_ids = quantity_move_line_ids
         return move_line_ids
 
-    def action_put_in_pack(self, **kwargs):
-        move_lines_to_pack = self._to_pack()
+    def action_put_in_pack(self, *, package_id=False, package_type_id=False, package_name=False):
```

Comment:
extra `*` argument?

Reply:
https://github.com/odoo/odoo/pull/203987#discussion_r2149331143
Kwargs stuff, should allow for further extensions in other modules with other specific kwargs? :thinking: 
Not 100% sure about this one, feel free to correct me.